src/vmm/tests/integration_tests.rs

```diff
@@ -178,3 +184,60 @@ fn test_pause_resume_microvm() {
         }
     }
 }
+
+#[test]
+#[cfg(target_arch = "x86_64")]
+fn test_create_snapshot() {
+    let snapshot_path = PathBuf::from("snapshot_file");
+    let mem_file_path = PathBuf::from("mem_file");
+
+    let pid = unsafe { libc::fork() };
+    match pid {
+        0 => {
+            set_panic_hook();
+            let boot_source_cfg: BootSourceConfig =
+                MockBootSourceConfig::new().with_default_boot_args().into();
+            let resources: VmResources = MockVmResources::new()
+                .with_boot_source(boot_source_cfg)
+                .into();
+            let mut event_manager = EventManager::new().unwrap();
+            let empty_seccomp_filter = get_seccomp_filter(SeccompLevel::None).unwrap();
+
+            let vmm = build_microvm(&resources, &mut event_manager, &empty_seccomp_filter).unwrap();
+
+            // Give the vCPUs a chance to attempt KVM_RUN.
+            thread::sleep(Duration::from_millis(200));
+
+            // Pause microVM.
+            vmm.lock().unwrap().pause_vcpus().unwrap();
+
+            // Create snapshot.
+            let snapshot_params = CreateSnapshotParams {
+                snapshot_type: SnapshotType::Full,
+                snapshot_path: snapshot_path.clone(),
+                mem_file_path: mem_file_path.clone(),
+                version: Some(1),
+            };
+
+            {
+                let mut locked_vmm = vmm.lock().unwrap();
+                persist::create_snapshot(&mut locked_vmm, snapshot_params).unwrap();
+            }
+
+            vmm.lock().unwrap().stop(0);
+        }
+        vmm_pid => {
+            // Parent process: wait for the vmm to exit.
+            let mut vmm_status: i32 = -1;
+            let pid_done = unsafe { libc::waitpid(vmm_pid, &mut vmm_status, 0) };
+            assert_eq!(pid_done, vmm_pid);
+            restore_stdin();
+            // If any panics occurred, its exit status will be != 0.
+            assert!(unsafe { libc::WIFEXITED(vmm_status) });
+            assert_eq!(unsafe { libc::WEXITSTATUS(vmm_status) }, 0);
```

Comment:
I think the parent should at least assert that the snapshot files aren't empty after the child has exited

Reply:
One step further would be for it to actually deserialize the state file and verify its contents

Reply:
Done.